vortex-array/src/lib.rs

```diff
@@ -65,7 +67,10 @@ pub mod flatbuffers {
 ///
 /// This is the main entrypoint for working with in-memory Vortex data, and dispatches work over the underlying encoding or memory representations.
 #[derive(Debug, Clone)]
-pub enum Array {
+pub struct Array(pub(crate) Inner);
+
+#[derive(Debug, Clone)]
+pub(crate) enum Inner {
```

Comment:
This split is the important part.

Adding/removing/changing one of the variants of a `pub enum` is a breaking change. This gives us more flexibility to evolve the implementation in the future